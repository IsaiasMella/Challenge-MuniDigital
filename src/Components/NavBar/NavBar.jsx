export const NavBar = ({ handleChange, handleModale }) => {
  return (
    <nav className="flex justify-between gap-4 bg-white py-4 px-8">
      <p className="font-bold text-3xl">
        to<span className="text-violet-600">do.</span>
      </p>
      <div className="flex flex-1 justify-center gap-10">
        <input
          className="outline-none bg-slate-100 rounded-2xl w-1/3 py-1 px-4 placeholder:text-slate-400 "
          type="text"
          name="search"
          onChange={(e) => {
            handleChange(e);
          }}
          placeholder="Buscar"
        />
        <select
          className="select-none outline-none border-b-[2px] border-slate-200 rounded-md  px-2"
          name="status"
          onChange={(e) => {
            handleChange(e);
          }}
        >
          <option value="all">Todas</option>
          <option value="hacer">Hacer</option>
          <option value="en curso">En curso</option>
          <option value="terminada">terminadas</option>
          <option value="rechazada">Rechazadas</option>
        </select>
      </div>
      <button onClick={handleModale} className="btn-primary flex-2">
        + Nueva tarea
      </button>
    </nav>
  );
};
