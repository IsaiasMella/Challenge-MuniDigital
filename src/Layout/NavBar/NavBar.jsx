export const NavBar = () => {
  return (
    <div className="py-4 px-10 w-full flex justify-between items-center text-slate-700">
      <p className="font-bold text-3xl">
        to<span className="text-violet-600">do.</span>
      </p>
      {/* <input
        className="outline-none bg-slate-100 rounded-2xl w-1/3 py-1 px-4 text-slate-500"
        type="text"
        placeholder="Buscar"
      /> */}
      <div>
        <p className="font-semibold text-xl">Isaias Mella</p>
      </div>
    </div>
  );
};
