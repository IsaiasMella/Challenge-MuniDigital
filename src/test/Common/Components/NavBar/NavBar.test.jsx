import { render, fireEvent } from '@testing-library/react';
import { NavBar } from '../../../../Components/NavBar/NavBar';

describe('<NavBar/>', () => {
  it('debe coincidir con el snapshot', () => {
    const { asFragment } = render(<NavBar handleChange={jest.fn()} handleModale={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('debe llamar a handleChange cuando se cambia el valor del input y select', () => {
    const handleChange = jest.fn();
    const { getByPlaceholderText, getByText } = render(<NavBar handleChange={handleChange} handleModale={jest.fn()} />);

    fireEvent.change(getByPlaceholderText('Buscar'), { target: { value: 'test' } });
    expect(handleChange).toHaveBeenCalled();

    fireEvent.change(getByText('Todas').parentNode, { target: { value: 'hacer' } });
    expect(handleChange).toHaveBeenCalledTimes(2);
  });

  it('debe llamar a handleModale cuando se hace clic en el botÃ³n', () => {
    const handleModale = jest.fn();
    const { getByText } = render(<NavBar handleChange={jest.fn()} handleModale={handleModale} />);

    fireEvent.click(getByText('+ Nueva tarea'));
    expect(handleModale).toHaveBeenCalled();
  });
});
