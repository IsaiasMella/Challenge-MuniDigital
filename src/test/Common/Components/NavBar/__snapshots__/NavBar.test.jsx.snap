// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> debe coincidir con el snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex justify-between gap-4 bg-white py-4 px-8"
  >
    <p
      class="font-bold text-3xl cursor-default"
    >
      to
      <span
        class="text-violet-600"
      >
        do.
      </span>
    </p>
    <div
      class="flex flex-1 justify-center gap-10"
    >
      <input
        class="outline-none bg-slate-100 rounded-2xl w-1/3 py-1 px-4 placeholder:text-slate-400 "
        name="search"
        placeholder="Buscar"
        type="text"
      />
      <select
        class="select-none outline-none cursor-pointer border-b-[2px] border-slate-200 rounded-md  px-2"
        name="status"
      >
        <option
          value="all"
        >
          Todas
        </option>
        <option
          value="hacer"
        >
          Hacer
        </option>
        <option
          value="en curso"
        >
          En curso
        </option>
        <option
          value="terminada"
        >
          terminadas
        </option>
        <option
          value="rechazada"
        >
          Rechazadas
        </option>
      </select>
    </div>
    <button
      class="btn-primary flex-2"
    >
      + Nueva tarea
    </button>
  </nav>
</DocumentFragment>
`;

exports[`Componente NavBar deber√≠a coincidir con el snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex justify-between gap-4 bg-white py-4 px-8"
  >
    <p
      class="font-bold text-3xl cursor-default"
    >
      to
      <span
        class="text-violet-600"
      >
        do.
      </span>
    </p>
    <div
      class="flex flex-1 justify-center gap-10"
    >
      <input
        class="outline-none bg-slate-100 rounded-2xl w-1/3 py-1 px-4 placeholder:text-slate-400 "
        name="search"
        placeholder="Buscar"
        type="text"
      />
      <select
        class="select-none outline-none cursor-pointer border-b-[2px] border-slate-200 rounded-md  px-2"
        name="status"
      >
        <option
          value="all"
        >
          Todas
        </option>
        <option
          value="hacer"
        >
          Hacer
        </option>
        <option
          value="en curso"
        >
          En curso
        </option>
        <option
          value="terminada"
        >
          terminadas
        </option>
        <option
          value="rechazada"
        >
          Rechazadas
        </option>
      </select>
    </div>
    <button
      class="btn-primary flex-2"
    >
      + Nueva tarea
    </button>
  </nav>
</DocumentFragment>
`;
